### PR TITLE
ntp: add config to set which interface ntp should listen

### PR DESCRIPTION
**Pull request by Pavan-Gunda**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind flake
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #https://github.com/elastisys/ck8s-ops/issues/5772

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: roles/kubernetes/preinstall/defaults/main.yml [hunk 1]
```diff
@@ -98,6 +98,9 @@ ntp_servers:
 ntp_restrict:
   - "127.0.0.1"
   - "::1"
+ntp_listen:
```
Comment (Xartos) [inline roles/kubernetes/preinstall/defaults/main.yml:101]:
```suggestion
ntp_interfaces:
```

File: roles/kubernetes/preinstall/templates/ntp.conf.j2 [hunk 2]
```diff
@@ -35,6 +35,11 @@ restrict -6 default kod notrap nomodify nopeer noquery limited
 restrict {{ item }}
 {% endfor %}
 
+interface ignore wildcard
+{% for item in ntp_listen %}
+interface listen {{ item }}
+{% endfor %}
```
Comment (Xartos) [inline roles/kubernetes/preinstall/templates/ntp.conf.j2:41]: To make it backwards compatible add a variable like `ntp_filter_interface` and set it to false and then:

```suggestion
{% if ntp_filter_interface %}
interface ignore wildcard
{% for item in ntp_listen %}
interface listen {{ item }}
{% endfor %}
{% endif %}
```

File: roles/kubernetes/preinstall/defaults/main.yml (hunk 1 above)
Comment (Xartos) [reply roles/kubernetes/preinstall/defaults/main.yml:101]: Maybe add some comments about the variables you add here as well

**Review by Xartos — APPROVED** [review]

**Review by lunkan93 — APPROVED** [review]

**Comment by Pavan-Gunda** [comment]
Closing this in favor of https://github.com/kubernetes-sigs/kubespray/pull/11066

**Pull request closed without merging**